lib/mutant.rb

```diff
@@ -193,22 +192,22 @@ class Config
 
     DEFAULT = new(
       debug:             false,
-      fail_fast:         false,
-      integration:       Integration::Null,
-      matcher:           Matcher::Config::DEFAULT,
-      includes:          EMPTY_ARRAY,
-      requires:          EMPTY_ARRAY,
-      isolation:         Mutant::Isolation::Fork,
-      reporter:          Reporter::CLI.build($stdout),
-      zombie:            false,
-      jobs:              Mutant.ci? ? CI_DEFAULT_PROCESSOR_COUNT : ::Parallel.processor_count,
       expected_coverage: Rational(1),
       expression_parser: Expression::Parser.new([
         Expression::Method,
         Expression::Methods,
         Expression::Namespace::Exact,
         Expression::Namespace::Recursive
-      ])
+      ]),
+      fail_fast:         false,
+      includes:          EMPTY_ARRAY,
+      integration:       Integration::Null,
+      isolation:         Mutant::Isolation::Fork,
+      jobs:              Mutant.ci? ? CI_DEFAULT_PROCESSOR_COUNT : ::Parallel.processor_count,
```

Comment:
> I remember we both liked the CI specific overwrite in the past, but I've changed my mind recently. I'll add a PR to kill it.

Yeah, I weigh the extra complexity vs our needs. Maybe when our CI was "static" it made sense, or maybe I just said "yeah sure" because we had other concerns to focus on. Now that we are moving CI elsewhere, to something a bit more dynamic, I would like to revisit this and make things simpler.

I *think* most of the time our new systems won't "lie" about how many cores are available, and when we can't make it report accurately we can just override it with explicit configuration.